WWST-1341 WWST-1342 DTH for Sensative Strips Comfort/Drip

greens opened the pull request.

greens committed:
WWST-1341 WWST-1342 DTH for Sensative Strips Comfort/Drip

greens committed:
WWST-1341 WWST-1342 DTH for Sensative Strips Comfort/Drip

greens commented:
Couldn't find another DTH where these fit any better.

greens committed:
WWST-1341 WWST-1342 DTH for Sensative Strips Comfort/Drip

MAblewicz commented on devicetypes/smartthings/zwave-water-temp-light-sensor.src/zwave-water-temp-light-sensor.groovy:
```diff
@@ -0,0 +1,226 @@
+}
+
+def installed() {
+	// Dome Leak Sensor sends WakeUpNotification every 12 hours. Please add zwaveinfo.mfr check when adding other sensors with different interval.
```
Dome?

greens replied:
just leftover from the `cp`

MAblewicz commented on devicetypes/smartthings/zwave-water-temp-light-sensor.src/zwave-water-temp-light-sensor.groovy:
```diff
@@ -0,0 +1,226 @@
+
+def updated() {
+	// Dome Leak Sensor sends WakeUpNotification every 12 hours. Please add zwaveinfo.mfr check when adding other sensors with different interval.
+	sendEvent(name: "checkInterval", value: (2 * 12 + 2) * 60 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
```
We started extracting send checkInterval event to one method. Feel free to do the same if you like. If something is wrong we need to change only one place.

MAblewicz commented on devicetypes/smartthings/zwave-water-temp-light-sensor.src/zwave-water-temp-light-sensor.groovy:
```diff
@@ -0,0 +1,226 @@
+				result << createEvent(descriptionText: "Water pressure $level", isStateChange: true)
+				break
+		}
+	} else if (cmd.notificationType == 0x04) {
```
Does that device report overheat?
Maybe consider using Temperature Alarm capability?

greens replied:
No. It doesn't.

But I figured it might be worth having this stuff for some hypothetical device in the future.

MAblewicz commented on devicetypes/smartthings/zwave-water-temp-light-sensor.src/zwave-water-temp-light-sensor.groovy:
```diff
@@ -0,0 +1,226 @@
+	def cmds = [ zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x01).format(),
+				zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x03).format(),
+				zwave.notificationV3.notificationGet(notificationType: 0x05).format(), //water alarm
+                zwave.batteryV1.batteryGet().format()]
```
Formating (spaces)

MAblewicz reviewed (dismissed):
Overall looks good. Only small changes may be included

mckeed commented on devicetypes/smartthings/zwave-water-temp-light-sensor.src/zwave-water-temp-light-sensor.groovy:
```diff
@@ -0,0 +1,226 @@
+		capability "Temperature Measurement"
+		capability "Illuminance Measurement"
+
+		fingerprint mfr: "019A", prod: "0003", model: "000A", deviceJoinName: "Sensative Strips"
```
There are three different sensors called "Strips" right? Shouldn't this be "Sensitive Strips Drip"

greens replied:
The Drip and the Comfort are literally the same device:

https://products.z-wavealliance.org/products/2942

there's no way to differentiate them. We're talking with Sensative about it now, but will probably just use this DTH for both. (Yes, they both detect water.)

greens committed:
Sensative Strips

dkirker commented on devicetypes/smartthings/zwave-water-temp-light-sensor.src/zwave-water-temp-light-sensor.groovy:
```diff
@@ -0,0 +1,228 @@
+			result << createEvent(descriptionText: "$device.displayName detected low temperature", isStateChange: true)
+		}
+	} else if (cmd.notificationType == 0x07) {
+		if (cmd.event == 0x03) {
```
Heh, I assume these devices don't actually support tamper? "here for the future"?

greens committed:
WWST-1341 WWST-1342 Fix cloud parsing

greens committed:
WWST-1341 WWST-1342 Two Sensative Strips DTHs

These devices can only be distinguished by their configuration values, so device type re-assignment has to happen after install.

greens commented:
Had to update this significantly to "distinguish" the two devices. The one thing that I can't solve is changing the install name, unfortunately. So both devices will still be called "Sensative Strips". Is there a way to do this @mckeed ?

greens committed:
WWST-1341 WWST-1342 Two Sensative Strips DTHs

These devices can only be distinguished by their configuration values, so device type re-assignment has to happen after install.

greens committed:
WWST-1341 WWST-1342 Two Sensative Strips DTHs

These devices can only be distinguished by their configuration values, so device type re-assignment has to happen after install.

shrakuma commented:
@greens sensative team said -  just spoke with the team and we like your recommendation! Please move forward with “Sensative Strips Comfort/Drip”
